### ISSUE TITLE
[Held requests] <WS> owes -0.00 displayed in LHN preview after partially approving report

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#43689]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.83-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y 
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
precondition: user is a member of a collect workspace with a set approver
1. Log in as a member on secondary device
2. Open the workspace chat and submit 2 expenses
3. Submit the report
4. On the primary device log in as an approver
5. Open the workspace chat
6. Select the submitted report
7. Put on hold an expense
8. Partially approve the report
9. Wait until the held expense gets moved to a new report
10. Navigate to the LHN

## Expected Result:
The correct amount that workspace owes is displayed in the LHN preview

## Actual Result:
<Workspace name> owes -$0.00 until the workspace chat is opened again

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/edef81b7-ecb8-4ecf-b22d-b8d46e5a6475

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01536daf4c62d44622</li>
        <li>Upwork Job ID: 1802724679894730679</li>
        <li>Last Price Increase: 2024-06-17</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @robertjchen</details>

**Comment by lanitochka17** [comment]
@garrettmknight  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by garrettmknight** [comment]
@robertjchen is this expected as part of the rollout? https://github.com/Expensify/Expensify/issues/274076

**Comment by robertjchen** [comment]
It's a bug with partial pay- we're working on this one here: https://github.com/Expensify/App/issues/41652

**Comment by jayeshmangwani** [comment]
@garrettmknight Should we hold/close this issue in favor of https://github.com/Expensify/App/issues/43689#issuecomment-2175801300 issue?

**Comment by robertjchen** [comment]
I think we can keep it open for now and re-test after the fixes are done there.

**Comment by garrettmknight** [comment]
Still holding

**Comment by garrettmknight** [comment]
Still holding on https://github.com/Expensify/App/issues/41652

**Comment by garrettmknight** [comment]
Still holding on https://github.com/Expensify/App/issues/41652

**Comment by garrettmknight** [comment]
Still holding on https://github.com/Expensify/App/issues/41652

**Comment by garrettmknight** [comment]
Still holding on https://github.com/Expensify/App/issues/41652

**Comment by garrettmknight** [comment]
@mallenexpensify can you keep an eye on this one next week while I'm OOO? Once https://github.com/Expensify/App/issues/41652 is done we can expect this to be off hold. I'll pick back up when I get back. 🤝

**Comment by mallenexpensify** [comment]
The PR we're held on hit staging 16 hours ago https://github.com/Expensify/App/pull/42573#issuecomment-2252008241

**Comment by jayeshmangwani** [comment]
@robertjchen After approving the partial payment, I can see `<WS> owes -0.00` for a second before it shows the correct value that the workspace owes.

Here's a clearer breakdown of the flow:

1. LHN Preview shows WS owes ₹1345.01
2. Approved ₹340.00
3. LHN shows WS owes $12.00
4. LHN shows WS owes ₹0.00
5. LHN shows WS owes ₹1005.01


https://github.com/user-attachments/assets/0ebaa701-09e8-412a-922d-2e0092a7b7ce

Sidenote: The same values are reflected in the IOU Preview in the report.

**Comment by mallenexpensify** [comment]
@jayeshmangwani the [PR we were holding on hit production yesterday](https://github.com/Expensify/App/pull/42573#issuecomment-2256646597) and your comment was 11 hours ago so I'm assuming you were on the latest version of production, comment if now.  

Let's see what Bobby Bottles has to say

**Comment by mallenexpensify** [comment]
Weird, commented but it didn't remove the `Overdue` label

**Comment by jayeshmangwani** [comment]
> your comment was 11 hours ago so I'm assuming you were on the latest version of production

Yes, I've tested in the latest version

**Comment by robertjchen** [comment]
Thanks for the latest tests- some discussion going on [Slack](https://expensify.slack.com/archives/C049HHMV9SM/p1722266811801429?thread_ts=1714761240.737509&cid=C049HHMV9SM) as well as to the source of the `0.00`. It appears to be coming from the backend- investigating further.

**Comment by garrettmknight** [comment]
@robertjchen Do you want to take this one on?

**Comment by robertjchen** [comment]
yep, I can take this one given that it's related to the other linked one that we're investigating.

**Comment by jayeshmangwani** [comment]
@robertjchen Should we make this Issue a weekly?

**Comment by robertjchen** [comment]
This is on the BE- it seems like the core issue is that `OpenReport` may not be not returning the total that incorporates the hold expenses.

**Comment by robertjchen** [comment]
A potential BE fix was deployed, let's re-test this!

**Comment by robertjchen** [comment]
Still reproducible- potential other cause, investigating.

https://github.com/user-attachments/assets/6e241147-171b-497c-b8bc-f54142b93b17

**Comment by robertjchen** [comment]
Still working on potential fix locally, it's a hairy one!

**Comment by robertjchen** [comment]
I fixed it!! 🎉 Going to get the changes reviewed and deployed

https://github.com/user-attachments/assets/3893aa73-a208-423e-842b-5007c2b8cf99

**Comment by robertjchen** [comment]
PR under review

**Comment by robertjchen** [comment]
Auth change deployed!
